VAMobile/src/api/letters/downloadLetter.tsx

```diff
@@ -0,0 +1,56 @@
+import FileViewer from 'react-native-file-viewer'
+
+import { useQuery } from '@tanstack/react-query'
+
+import { LetterTypes, LettersDownloadParams } from 'api/types'
+import { Events, UserAnalytics } from 'constants/analytics'
+import { Params } from 'store/api'
+import { DEMO_MODE_LETTER_ENDPOINT, DEMO_MODE_LETTER_NAME } from 'store/api/demo/letters'
+import { logAnalyticsEvent, setAnalyticsUserProperty } from 'utils/analytics'
+import getEnv from 'utils/env'
+import { downloadDemoFile, downloadFile } from 'utils/filesystem'
+import { registerReviewEvent } from 'utils/inAppReviews'
+
+import { lettersKeys } from './queryKeys'
+
+const { API_ROOT } = getEnv()
+
+/**
+ * Fetch user letter
+ */
+const downloadLetter = async (
+  letterType: LetterTypes,
+  demoMode: boolean,
+  lettersOption: LettersDownloadParams,
+): Promise<boolean | undefined> => {
+  const lettersAPI = `${API_ROOT}/v0/letters/${letterType}/download`
+  const filePath = demoMode
+    ? await downloadDemoFile(DEMO_MODE_LETTER_ENDPOINT, DEMO_MODE_LETTER_NAME, lettersOption as unknown as Params)
+    : await downloadFile('POST', lettersAPI, `${letterType}.pdf`, lettersOption as unknown as Params, 3)
+  if (filePath) {
+    registerReviewEvent()
+    logAnalyticsEvent(Events.vama_letter_download(letterType))
+    setAnalyticsUserProperty(UserAnalytics.vama_uses_letters())
+    await FileViewer.open(filePath)
+    return true
+  }
+}
+
+/**
+ * Returns a query for a user letter
+ */
+export const useDownloadLetter = (
+  letterType: LetterTypes,
+  demoMode: boolean,
```

Comment:
Instead of passing `demoMode` as a param, you can directly access the store, and check for it that way

```tsx
store.getState().demo.demoMode
```

Reply:
updated